zebra-chain/src/transaction.rs

```diff
@@ -1,5 +1,7 @@
 //! Transactions and transaction-related structures.
 
+use futures::future::Either;
```

Comment:
Let's replace `Either<...>` with a named and typed enum.

We could do something like:
```rust
enum JoinSplit<'_> {
    JoinSplitOnBctv14(&'_ JoinSplitData<Bctv14Proof>),
    JoinSplitOnGroth16(&'_ JoinSplitData<Groth16Proof>),
}
```

You'll need lifetimes here, the compiler should give you hints if I've got it wrong.